hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1346,6 +1348,17 @@ public void startLoading() {
         }
     }
 
+    private void addLoadingFuture(Future<?> e) {
+        if (e instanceof CompletableFuture) {
+            ((CompletableFuture<?>) e).whenCompleteAsync((result, throwable) -> {
```

Comment:
Can it not be a CompletableFuture? If not, maybe we should change signatures so that it's guaranteed by the signature